salalib/axialpolygons.cpp

```diff
@@ -244,24 +229,14 @@ void AxialPolygons::makePixelPolys()
    int i = 0;
 
    // record all of this onto the pixel polygons
-   if (m_pixel_polys)
-   {
-      for (i = 0; i < m_cols; i++) {
-         delete [] m_pixel_polys[i];
-      }
-      delete [] m_pixel_polys;
-      m_pixel_polys = NULL;
-   }
-   m_pixel_polys = new pvecint *[m_cols];
-   for (i = 0; i < m_cols; i++) {
-      m_pixel_polys[i] = new pvecint[m_rows];
-   }
+
+   m_pixel_polys = std::vector<std::vector<int> > (m_cols*m_rows);
```

Comment:
This should be a `ColumnMatrix<std::vector<int>>(m_rows, m_cols)`. For accessing the elem (x,y) you jump x * m_rows, plus y, i.e. you jump x columns to the right (each column being m_rows elements long), and then walk down y rows to the target cell in the column.

Reply:
There might be a performance impact depending on how the x and y in the underlying data are distributed. I.e. if all y's for one x come first, you chose the correct matrix. If all x's for one y come first, you want to flip the matrix.

Reply:
Give me a shout if you want to discuss the how and why...